### PR TITLE
Create song of the week

### PR DESCRIPTION
**Pull request by jesseez**
Adds the first section for the song of the week.

1. Adds a new data file (and add some utils)
2. Add the card renderer, attribution renderer, and section content
3. peepeepoopoo
4. Makes it super pretty and responsive. Huge shoutout to Vedant

image

image

image

### CODE REVIEW

File: web/src/components/home/subcomponents/song_of_the_week/messages.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const messages = {
+  title: "Song of the Week",
+  listenOnYoutube: "YouTube",
+  listenOnSpotify: "Spotify",
+};
```
Comment (Kuboosh21) [inline web/src/components/home/subcomponents/song_of_the_week/messages.js:5]: why do you separate these in to a separate file if they're so small and not re-used anywhere? That seems a bit overengineered

File: web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js [hunk 2]
```diff
@@ -1,5 +1,43 @@
+import { Box, Card, Container, Stack, Typography } from "@mui/material";
+import songOfTheWeekStyles from "./song_of_the_week.styles";
+import SongCard from "./song_card";
+import getCurrentSong from "../../../../data/songs_of_the_week";
```
Comment (Kuboosh21) [inline web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js:4]: can this path be shortened?

File: web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js [hunk 3]
```diff
@@ -1,5 +1,43 @@
...
+import messages from "./messages";
+import QuoteAttribution from "./quote_attribution";
+
 function SongOfTheWeek() {
-  return <div>Song of the week</div>;
+  let songInfo = getCurrentSong();
+  console.log(songInfo);
```
Comment (Kuboosh21) [inline web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js:10]: remove console log

File: web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.styles.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    backgroundColor: theme.palette.background.lightGray,
+    padding: "24px",
+  },
+  songOfTheWeekFlexContainer: {
+    xs: "column",
+    md: "row",
+  },
+  title: {
+    paddingBottom: "8px",
+    // TODO: update the titles to be bold as per Vedant's suggestion
```
Comment (Kuboosh21) [inline web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.styles.js:14]: is this a TODO for now or later?

File: web/src/data/songs_of_the_week.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      quote:
+        "This song is one of my favorites to dance to. " +
+        "This month's creative expressions class will be working on choreography to this song. " +
+        "Absolute banger, and brings back a lot of great memories.",
+      attributionInfo: attributors.riley,
+    },
+    {
+      name: "Old School Turntables",
+      artist: "Matt DiMona",
+      year: "2017",
```
Comment (Kuboosh21) [inline web/src/data/songs_of_the_week.js:35]: any way we could pull this from the Spotify API instead of hardcoding it here?

File: web/src/components/home/subcomponents/song_of_the_week/messages.js (hunk 1 above)
Comment (jesseez) [reply web/src/components/home/subcomponents/song_of_the_week/messages.js:5]: You seem a bit overengineered

File: web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.styles.js (hunk 4 above)
Comment (jesseez) [reply web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.styles.js:14]: For later

File: web/src/data/songs_of_the_week.js (hunk 5 above)
Comment (jesseez) [reply web/src/data/songs_of_the_week.js:35]: feel free to take a look at https://developer.spotify.com/documentation/web-api

TLDR is not simple. Way easier to just hardcode

File: web/src/components/home/subcomponents/song_of_the_week/messages.js (hunk 1 above)
Comment (jesseez) [reply web/src/components/home/subcomponents/song_of_the_week/messages.js:5]: Consistency really. Separation of concerns (template vs text vs styling)

File: web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js (hunk 3 above)
Comment (jesseez) [reply web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js]: Good catch, done

File: web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js (hunk 2 above)
Comment (jesseez) [reply web/src/components/home/subcomponents/song_of_the_week/song_of_the_week.js]: Good catch, done.

File: web/src/data/songs_of_the_week.js (hunk 5 above)
Comment (Kuboosh21) [reply web/src/data/songs_of_the_week.js:35]: u right

**Review by Kuboosh21 — APPROVED** [review]

File: web/src/data/songs_of_the_week.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const songs = [
+  new SongOfTheWeek(
+    createDate("05/26/2024"),
+    {
+      quote:
+        "This song is one of my favorites to dance to. " +
+        "This month's creative expressions class will be working on choreography to this song. " +
+        "Absolute banger, and brings back a lot of great memories.",
```
[the remark this answers is not in the document]
Comment (jesseez) [reply web/src/data/songs_of_the_week.js:29]: Literally came from Riley. You can even ask them
